### PR TITLE
Change Enum to Python 3

### PR DESCRIPTION
**Pull request by matthewfcarlson**
Python 3.4+ includes support for real enums. 
This class is offered as a compatibility shim.

### CODE REVIEW

**Comment by spbrogan** [comment]
since we know the users can we just convert them over to EnumInt or the std Enum.  I am not sure the feature is even used anymore anyway.

**Comment by matthewfcarlson** [comment]
I just added it for the sake of backwards compatibility

File: edk2toollib/utility_functions_test.py [hunk 1]
```diff
@@ -65,5 +65,32 @@ def test_locate_class_in_module(self):
...
+        self.assertFalse(hasattr(animals, "dog"))
+        # check to make sure the values are unique
+        self.assertNotEqual(animals.Dog, animals.Cat)
+
+        # make sure we can iterate over the members
+        for animal in animals.__members__:
+            self.assertIn(animal, all_animals)
+
+
+    def test_EnumClass_args(self):
```
Comment (corthon) [inline edk2toollib/utility_functions_test.py]: Can we add a test for "not in enum"?

**Comment by corthon** [comment]
I agree with the "let's move over to the native thing". Is there some way we could keep this back-compat PR, but spit out a warning to get people to migrate so that we can drop this in the next major version? @spbrogan, would that even be desirable?

File: edk2toollib/utility_functions_test.py (hunk 1 above)
Comment (matthewfcarlson) [reply edk2toollib/utility_functions_test.py:86]: We do have a test to make sure it's not in.
``` python
self.assertFalse(hasattr(animals, "Rat"))
self.assertFalse(hasattr(animals, "dog"))
```

We could create two seperate enums and check to make sure they aren't in each other

**Comment by matthewfcarlson** [comment]
I think that would be pretty easy to do, since we have our own constructor. I like Bret's solution.

File: edk2toollib/utility_functions_test.py (hunk 1 above)
Comment (corthon) [reply edk2toollib/utility_functions_test.py:86]: Didn't notice that.

**Pull request merged**